webapp/channels/src/components/advanced_text_editor/use_key_handler.tsx

```diff
@@ -0,0 +1,368 @@
+            caretPosition,
+        );
+
+        if (ignoreKeyPress) {
+            e.preventDefault();
+            e.stopPropagation();
+            return;
+        }
+
+        if (allowSending && isValidPersistentNotifications) {
+            e.persist?.();
+
+            // textboxRef.current?.blur();
+
+            if (withClosedCodeBlock && message) {
+                handleSubmit(e, {...draft, message});
+            } else {
+                handleSubmit(e);
+            }
+
+            // setTimeout(() => {
+            //     focusTextbox();
+            // });
+        }
+
+        emitTypingEvent();
+    }, [draft, ctrlSend, codeBlockOnCtrlEnter, caretPosition, postId, emitTypingEvent, handleSubmit, isValidPersistentNotifications]);
+
+    const handleKeyDown = useCallback((e: React.KeyboardEvent<TextboxElement>) => {
+        const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
+        const ctrlEnterKeyCombo = (ctrlSend || codeBlockOnCtrlEnter) &&
+            Keyboard.isKeyPressed(e, KeyCodes.ENTER) &&
+            ctrlOrMetaKeyPressed;
+
+        const ctrlKeyCombo = Keyboard.cmdOrCtrlPressed(e) && !e.altKey && !e.shiftKey;
+        const ctrlAltCombo = Keyboard.cmdOrCtrlPressed(e, true) && e.altKey;
+        const shiftAltCombo = !Keyboard.cmdOrCtrlPressed(e) && e.shiftKey && e.altKey;
+        const ctrlShiftCombo = Keyboard.cmdOrCtrlPressed(e, true) && e.shiftKey;
+
+        // fix for FF not capturing the paste without formatting event when using ctrl|cmd + shift + v
+        if (e.key === KeyCodes.V[0] && ctrlOrMetaKeyPressed) {
+            if (e.shiftKey) {
+                isNonFormattedPaste.current = true;
+                timeoutId.current = window.setTimeout(() => {
+                    isNonFormattedPaste.current = false;
+                }, 250);
+            }
+        }
+
+        // listen for line break key combo and insert new line character
+        if (Utils.isUnhandledLineBreakKeyCombo(e)) {
+            handleDraftChange({
+                ...draft,
+                message: Utils.insertLineBreakFromKeyEvent(e.nativeEvent),
+            });
+            return;
+        }
+
+        if (ctrlEnterKeyCombo) {
+            postMsgKeyPress(e);
+            return;
+        }
+
+        if (Keyboard.isKeyPressed(e, KeyCodes.ESCAPE)) {
+            textboxRef.current?.blur();
+        }
+
+        const upKeyOnly = !ctrlOrMetaKeyPressed && !e.altKey && !e.shiftKey && Keyboard.isKeyPressed(e, KeyCodes.UP);
+        const messageIsEmpty = draft.message.length === 0;
+        const allowHistoryNavigation = draft.message.length === 0 || draft.message === messageHistory[messageHistoryIndex.current];
+        const caretIsWithinCodeBlock = caretPosition && isWithinCodeBlock(draft.message, caretPosition); // REVIEW
+
+        if (upKeyOnly && messageIsEmpty) {
+            e.preventDefault();
+            if (textboxRef.current) {
+                textboxRef.current.blur();
```

Comment:
Similarly, I don't know why we'd need to blur this when it'd happen automatically when the edit box focuses itself on show. We don't need to change it now though since there's enough happening here already